pkg/distributor/distributor.go

```diff
@@ -1161,6 +1220,15 @@ func (d *Distributor) prePushHaDedupeMiddleware(next PushFunc) PushFunc {
 
 		numSamples := 0
 		now := time.Now()
+
+		getReplicaForSample := func(i int) haReplica {
+			cluster, replica := findHALabels(haReplicaLabel, haClusterLabel, req.Timeseries[i].Labels)
```

Comment:
wdyt about changing `findHALabels` to always return an `haReplica`? that should remove the need for `getReplicaForSample`